src/main/java/cubicchunks/worldgen/gui/component/UIBlockStateButton.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.worldgen.gui.component;
+
+import java.util.Map.Entry;
+import java.util.function.Consumer;
+
+import org.lwjgl.opengl.GL11;
+
+import cubicchunks.worldgen.gui.DummyWorld;
+import net.malisis.core.client.gui.GuiRenderer;
+import net.malisis.core.client.gui.MalisisGui;
+import net.malisis.core.client.gui.component.UIComponent;
+import net.minecraft.block.properties.IProperty;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.VertexBuffer;
+import net.minecraft.client.renderer.texture.ITextureObject;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+
+public class UIBlockStateButton extends UIComponent<UIBlockStateButton> {
+
+    private IBlockState iBlockState;
+    private final Consumer<IBlockState> onClick;
+
+    public UIBlockStateButton(MalisisGui gui, IBlockState iBlockState1, Consumer<IBlockState> clickAction) {
```

Comment:
Instead of Consumer as argument you take Runnable, and then suppose you construct `UIBlockStateButton` like this:

```java
IBlockState state = getSomeState();
UIBlockStateButton btn = new UIBlockStateButton(gui, state, s -> doSomethingWithState(s));
```

then you change it to:
```java
IBlockState state = getSomeState();
UIBlockStateButton btn = new UIBlockStateButton(gui, state, () -> doSomethingWithState(state));
```

And in UIBlockStateButton instead of `consumer.accept(state)` you do `runnable.run()`